packages/components/src/notice/index.native.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * External dependencies
+ */
+import {
+	Animated,
+	Easing,
+	Text,
+	TouchableWithoutFeedback,
+	View,
+	Dimensions,
+} from 'react-native';
+
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useRef, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import styles from './style.scss';
+
+const Notice = ( { onNoticeHidden, content, id } ) => {
+	const [ width, setWidth ] = useState( Dimensions.get( 'window' ).width );
+	const [ visible, setVisible ] = useState( true );
+
+	const animationValue = useRef( new Animated.Value( 1 ) ).current;
+	const timer = useRef( null );
+
+	const onDimensionsChange = () => {
+		setWidth( Dimensions.get( 'window' ).width );
+	};
+
+	Dimensions.addEventListener( 'change', onDimensionsChange );
```

Comment:
I am not sure if this is a good idea to call this here, wdyt? @geriux

Reply:
We should move this to be within a `useEffect`. We can't use the one we have now because it will be triggered everytime `visible` changes so let's create another `useEffect` that will contain this line and also that will return a callback to remove the listener.

Something like:

```suggestion
	useEffect( () => {
		Dimensions.addEventListener( 'change', onDimensionsChange );
		return () => {
			Dimensions.removeEventListener( 'change', onDimensionsChange );
		};
	}, [] );
```

What do you think?

Reply:
Aha I knew this wasn't ideal! Thanks